pkg/resources/table_acceptance_test.go

```diff
@@ -1475,6 +1480,158 @@ func TestAcc_Table_MaskingPolicy(t *testing.T) {
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Table_DefaultDataRetentionTime/WithoutTable"),
+				ConfigVariables: configVariablesWithoutTableDataRetentionTime(5),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_table.test", "data_retention_time_in_days", "-1"),
+					checkSchemaAndTableDataRetentionTime(id, 5, 5),
+				),
+			},
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Table_DefaultDataRetentionTime/WithoutTable"),
+				ConfigVariables: configVariablesWithoutTableDataRetentionTime(10),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_table.test", "data_retention_time_in_days", "-1"),
+					checkSchemaAndTableDataRetentionTime(id, 10, 10),
+				),
+			},
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Table_DefaultDataRetentionTime/WithTable"),
+				ConfigVariables: configVariablesWithTableDataRetentionTime(10, 5),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_table.test", "data_retention_time_in_days", "5"),
+					checkSchemaAndTableDataRetentionTime(id, 10, 5),
+				),
+			},
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Table_DefaultDataRetentionTime/WithTable"),
+				ConfigVariables: configVariablesWithTableDataRetentionTime(10, 15),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_table.test", "data_retention_time_in_days", "15"),
+					checkSchemaAndTableDataRetentionTime(id, 10, 15),
+				),
+			},
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Table_DefaultDataRetentionTime/WithoutTable"),
+				ConfigVariables: configVariablesWithoutTableDataRetentionTime(10),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_table.test", "data_retention_time_in_days", "-1"),
+					checkSchemaAndTableDataRetentionTime(id, 10, 10),
+				),
+			},
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Table_DefaultDataRetentionTime/WithTable"),
+				ConfigVariables: configVariablesWithTableDataRetentionTime(10, 0),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_table.test", "data_retention_time_in_days", "0"),
+					checkSchemaAndTableDataRetentionTime(id, 10, 0),
+				),
+			},
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Table_DefaultDataRetentionTime/WithTable"),
+				ConfigVariables: configVariablesWithTableDataRetentionTime(10, 3),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_table.test", "data_retention_time_in_days", "3"),
+					checkSchemaAndTableDataRetentionTime(id, 10, 3),
+				),
+			},
+		},
+	})
+}
+
+// proves https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2356 issue is fixed.
+func TestAcc_Table_DefaultDataRetentionTime_SetOutsideOfTerraform(t *testing.T) {
```

Comment:
because table is at the bottom of this hierarchy, can we have a test where we clean the above objects (i.e. we clean schema - it goes to db, we later clean db - it goes to account). Just for the sake of showing the behavior in the most complex scenario. (not needed but nice)

Reply:
I just asked for one additional test with the bigger hierarchy, not to change the existing one. But I guess it's also fine to leave it as is.